what4/test/Abduct.hs

```diff
@@ -0,0 +1,158 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Main where
+
+import           Test.Tasty
+import           Test.Tasty.HUnit
+
+import Data.Foldable (forM_)
+import Data.List (length)
+import System.IO (FilePath, IOMode(..), openFile, hClose)
+import System.IO.Temp (withSystemTempFile)
+import Data.Text
+import Data.Parameterized.Nonce (newIONonceGenerator)
+import Data.Parameterized.Some (Some(..))
+
+import What4.Config (extendConfig)
+import What4.Expr
+         ( ExprBuilder,  FloatModeRepr(..), newExprBuilder
+         , BoolExpr, IntegerExpr, GroundValue, groundEval
+         , EmptyExprBuilderState(..))
+import What4.Interface
+         ( BaseTypeRepr(..), getConfiguration
+         , freshConstant, safeSymbol, notPred
+         , impliesPred, intLit, intAdd, intLe )
+import What4.Solver
+import What4.Symbol (SolverSymbol(..))
+import What4.Protocol.SMTLib2 as SMT2
+         (assume, sessionWriter, runCheckSat, runGetAbducts, Writer)
+import What4.Protocol.SMTWriter
+         (mkSMTTerm)
+import What4.Protocol.Online
+
+cvc5executable :: FilePath
+cvc5executable = "cvc5"
+
+-- Call the online getAbduct tactic
+testGetAbductOnline ::
+  ExprBuilder t st fs ->
+  [BoolExpr t] ->
+  BoolExpr t ->
+  Int ->
+  IO [String]
+testGetAbductOnline sym hs g n = do
+  -- Print SMT file in /tmp/
+  withSystemTempFile "what4abdonline" $ \fname mirroredOutput -> do
+    proc <- startSolverProcess @(SMT2.Writer CVC5) cvc5Features (Just mirroredOutput) sym
+    let conn = solverConn proc
+    inNewFrame proc $ do
+      mapM_ (\x -> assume conn x) hs
+      getAbducts proc n (pack "abd") g
+    -- hClose mirroredOutput
```

Comment:
Can this line be removed? I would expect this to be unnecessary since `withSystemTempFile` should close the handle for you.